scripts/build-job-matrix.py

```diff
@@ -0,0 +1,62 @@
+import json
+import os
+import sys
+from itertools import product
+
+def load_config(config_path, default_config=None):
+    """Load JSON configuration from the given path."""
+    try:
+        with open(config_path, 'r') as file:
+            config = json.load(file)
+            if not config and default_config is not None:
+                return default_config
+            return config
+    except (FileNotFoundError, json.JSONDecodeError):
+        if default_config is not None:
+            return default_config
+        else:
+            raise
+
+def main(image_dirs):
+    print("Image directories:", image_dirs)  
+
+    default_config_path = 'opensciencegrid/default-build-config.json'
+    default_config = load_config(default_config_path)
+
+    include_list = []
+
+    for image_dir in image_dirs:
+        if not os.path.isdir(image_dir):
+            sys.exit(f"Error: Image directory '{image_dir}' does not exist.")
+        build_config_path = os.path.join(image_dir, 'build-config.json')
+
+        config = load_config(build_config_path, default_config)
+
+        image_name = os.path.basename(image_dir)
+
+        base_os_list = config['base_os'][0].split(', ')
+        osg_series_list = config['osg_series'][0].split(', ')
```

Comment:
It looks like you had to grab the first item in the list and split on the `,` because in the `s3-backup` and `sssd` configs, you were storing the "lists" as a single comma-delimited string as the first (and only) item in the lists. You could make these look like line 39 if you forced the config to be actual lists.